packages/flutter_tools/templates/module/android/library_new_embedding/include_flutter.groovy.copy.tmpl

```diff
@@ -1,5 +1,13 @@
-def scriptFile = getClass().protectionDomain.codeSource.location.toURI()
-def flutterProjectRoot = new File(scriptFile).parentFile.parentFile
+def gradle = null
+def flutterProjectRoot = null
+if (!getBinding().getVariables().containsKey("gradle")) {
+    gradle = this
+    flutterProjectRoot = gradle.buildscript.getSourceFile().getParentFile().getParentFile().absolutePath
+} else {
+    gradle = getBinding().getVariables().get("gradle")
+    def scriptFile = getClass().protectionDomain.codeSource.location.toURI()
+    flutterProjectRoot = new File(scriptFile).parentFile.parentFile.absolutePath
+}
```

Comment:
I think a comment here would be nice, describing when each case happens.

Reply:
Good idea, added!